src/components/uploadPicture/uploadImage.tsx

```diff
@@ -0,0 +1,199 @@
+import React, { useState, useEffect, useCallback, useRef } from "react";
+import styles from "./uploadImage.module.css";
+import { trpc } from "@/client/lib/trpc";
+import axios from "axios";
+import CameraIcon from "@mui/icons-material/PhotoCamera";
+import CheckCircleIcon from "@mui/icons-material/CheckCircle";
+import CircularProgress from "@mui/material/CircularProgress";
+import Typography from "@mui/material/Typography";
+
+interface UploadImageComponentProps {
+  title: string;
+  fileName: string;
+  onUpload: (fileId: string) => void;
+}
+
+enum UploadStep {
+  Start,
+  TakePicture,
+  UploadPicture,
+}
+
+const UploadImageComponent: React.FC<UploadImageComponentProps> = ({
+  title,
+  fileName,
+  onUpload,
+}) => {
+  const [currentStep, setCurrentStep] = useState(UploadStep.Start);
+  const [uploading, setUploading] = useState(false);
```

Comment:
this can be derived, something like
```ts
const isUploading = getPresignedUrlForUpload.isLoading || finishFileUpload.isLoading
```

Reply:
I am not sure how to get rid of the currentStep states with isUploading state which has only two values, true or false.

Reply:
sorry, I meant `uploading` can be derived and `currentStep` can be kept as-is